### PR TITLE
Refactor syntax unicode ep constructors1

### PR DESCRIPTION
**Pull request by livecodepanos**
Updates on
`
bool MCButton::selectedchunk(MCStringRef& r_string)
bool MCButton::selectedline(MCStringRef& r_string)
bool MCField::returnchunk(int4 p_si, int4 p_ei, MCStringRef& r_chunk)
bool MCField::returnline(int4 si, int4 ei, MCStringRef& r_string)
`

### CODE REVIEW

File: engine/src/button.cpp [hunk 1]
```diff
@@ -3782,9 +3782,11 @@ void MCButton::setupmenu()
 bool MCButton::selectedchunk(MCStringRef& r_string)
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint4 number = ep.getuint4();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
+	uint4 number = (uint4) MCNumberFetchAsUnsignedInteger(*t_number);
```
Comment (livecodesebastien) [inline engine/src/button.cpp]: You should check whether `ctxt.HasError()` after calling `getnumberprop()`and  before fetching `t_number` - or mark it as `/* UNCHECKED */`

File: engine/src/button.cpp [hunk 2]
```diff
@@ -3794,10 +3796,10 @@ bool MCButton::selectedchunk(MCStringRef& r_string)
...
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint2 number;
-	ep.getuint2(number, 0, 0, EE_UNDEFINED);
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
+	uint2 number = (uint2) MCNumberFetchAsUnsignedInteger(*t_number);
```
Comment (livecodesebastien) [inline engine/src/button.cpp]: Same here

File: engine/src/fields.cpp [hunk 3]
```diff
@@ -2263,9 +2264,10 @@ Boolean MCField::selectedmark(Boolean whole, int4 &si, int4 &ei,
 bool MCField::returnchunk(int4 p_si, int4 p_ei, MCStringRef& r_chunk)
 {
 	MCExecPoint ep(nil, nil, nil);
-	getprop(0, P_NUMBER, ep, False);
-	ep.ton();
-	uint2 number = ep.getuint2();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
+	uint2 number = (uint2) MCNumberFetchAsUnsignedInteger(*t_number);
```
Comment (livecodesebastien) [inline engine/src/fields.cpp]: Same

File: engine/src/fields.cpp [hunk 4]
```diff
@@ -2279,9 +2281,10 @@ bool MCField::returnchunk(int4 p_si, int4 p_ei, MCStringRef& r_chunk)
 bool MCField::returnline(int4 si, int4 ei, MCStringRef& r_string)
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint2 number = ep.getuint2();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
+	uint2 number = (uint2) MCNumberFetchAsUnsignedInteger(*t_number);
```
Comment (livecodesebastien) [inline engine/src/fields.cpp]: Same here

File: engine/src/button.cpp [hunk 5]
```diff
@@ -3782,9 +3782,10 @@ void MCButton::setupmenu()
 bool MCButton::selectedchunk(MCStringRef& r_string)
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint4 number = ep.getuint4();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	/* UNCHECKED */ getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
```
Comment (runrevmark) [inline engine/src/button.cpp]: Use getuintprop here instead (no need to make a valueref).

File: engine/src/button.cpp [hunk 6]
```diff
@@ -3794,10 +3795,10 @@ bool MCButton::selectedchunk(MCStringRef& r_string)
 bool MCButton::selectedline(MCStringRef& r_string)
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint2 number;
-	ep.getuint2(number, 0, 0, EE_UNDEFINED);
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	/* UNCHECKED */ getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
```
Comment (runrevmark) [inline engine/src/button.cpp]: Use getuintprop here.

File: engine/src/fields.cpp [hunk 7]
```diff
@@ -2263,9 +2264,10 @@ Boolean MCField::selectedmark(Boolean whole, int4 &si, int4 &ei,
 bool MCField::returnchunk(int4 p_si, int4 p_ei, MCStringRef& r_chunk)
 {
 	MCExecPoint ep(nil, nil, nil);
-	getprop(0, P_NUMBER, ep, False);
-	ep.ton();
-	uint2 number = ep.getuint2();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	/* UNCHECKED */ getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
```
Comment (runrevmark) [inline engine/src/fields.cpp]: use getuintprop here.

File: engine/src/fields.cpp [hunk 8]
```diff
@@ -2279,9 +2281,10 @@ bool MCField::returnchunk(int4 p_si, int4 p_ei, MCStringRef& r_chunk)
 bool MCField::returnline(int4 si, int4 ei, MCStringRef& r_string)
 {
 	MCExecPoint ep(nil, nil, nil);
-	/* UNCHECKED */ getprop(0, P_NUMBER, ep, False);
-	/* UNCHECKED */ ep.ton();
-	uint2 number = ep.getuint2();
+	MCExecContext ctxt(ep);
+	MCAutoNumberRef t_number;
+	/* UNCHECKED */ getnumberprop(ctxt, 0, P_NUMBER, False, &t_number);
```
Comment (runrevmark) [inline engine/src/fields.cpp]: use getuintprop here.

**Comment by runrevmark** [comment]
Hi Panos - this is looking good - could you merge in refactor-syntax_unicode again - thanks!

**Pull request merged**